### DISCUSSION TITLE
why is one case in switch cases repeated?

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#4417] [Q&A]
So I have a C script which look like this:
``` #include <stdio.h>
 
int main() {
 
 int digit; 
 
 printf("Enter digit:");
 scanf("%d",&digit); //Input digit
 
 switch (digit) { //Writing a case for every digit in 0-9
 
   case 1:
           printf("One\n");
           break;
   case 2:
           printf("Two\n");
           break;
   case 3:
           printf("Three\n");
           break;
   case 4:
           printf("Four\n");
           break;
   case 5:
           printf("Five\n");
           break;
   case 6:
           printf("Six\n");
           break;
   case 7:
         
   case 8:
           printf("Eight\n");
           break;
   case 9:
   case 10:
	   printf("ten\n");
	   
		  
   case 0:
           printf("Zero\n");
           break;
   default:
           printf("Invalid Digit\n");
 }
 
 return 0;
}
 ```
And Now my case 7 and case 9 are repeated in getCases() method. Why is that?
I am trying to find out, which case don't have body/ or are empty like case 7 and 9 above.
My code snippets:
``Jtables = res.getHighFunction().getJumpTables() 
 Jtables[0].getCases()[0]``

**Comment by Tim---** [comment]
If you have a fall-through with no instructions, compilers will generally generate the same jump address for both cases. From what I understand, Ghidra uses this fact to show the fall-through in the decompiled code, merging the cases that jump to the same address.

Apparently you don't have this behavior so the first question that comes to mind is: are the cases 7 and 8 jumping to the same address ?
If not, what are the instructions in both cases ?
